app/javascript/components/batch_update/components/BatchUpdateForm.js

```diff
@@ -266,38 +310,25 @@ class BatchUpdateForm extends React.Component {
           </Button>
         </Controls>
 
-        <h3>Edit Genes</h3>
-
-        <Genes>
-          {geneNames.map(name => (
-            <GeneInput
-              key={name}
-              name={name}
-              value={geneValues[name]}
-              onChangeValue={this.onChangeGeneValue}
-            />
-          ))}
-        </Genes>
-
-        {geneNames.length === 0 && (
-          <EmptyGenesMessage>
-            There aren’t any genes that describe all of your selected works
-          </EmptyGenesMessage>
-        )}
+        <Sans size="8" mt={3} mb={1}>
+          Tags
+        </Sans>
+
+        {this.renderTagInputs()}
+
+        <TagAutosuggest placeholder="Add a tag" onSelectTag={this.onAddTag} />
+
+        <Sans size="8" mt={4} mb={1}>
+          Genes
+        </Sans>
+
+        {this.renderGeneInputs()}
```

Comment:
These input rendering stanzas were really adding up, so I refactored them into the functions above.